packages/api/src/shared/GenericParser.ts

```diff
@@ -0,0 +1,142 @@
+import fs from "fs";
+import path from "path";
+import xlsx from "node-xlsx";
+import csvSyncParser = require("csv-parse/sync");
+import {
+    BeforeAdaptation,
+    DefaultObject,
+    NestedBeforeAdaptation,
+    NestedDefaultObject,
+    ParserInfo,
+    ParserPath,
+} from "../@types";
+import type { ValueWithPath } from "./@types/ValueWithPath";
+
+export class GenericParser {
+    /*
+        from parser info and original data, returns original key and data before adaptation
+    */
+    static findValueAndOriginalKeyByPath<T extends BeforeAdaptation>(
+        data: NestedBeforeAdaptation<T>,
+        parserData: ParserPath | ParserInfo<T, unknown>,
+    ): ValueWithPath<T> | undefined {
+        type TypeIn = T | DefaultObject;
+        const mapper: ParserPath = Array.isArray(parserData) ? parserData : parserData.path;
+        const successiveKeys: string[] = [];
+        let objectToLookIn: TypeIn = data;
+        let oneLevelKey: string | undefined;
+
+        for (const possibleKeys of mapper) {
+            // If it is a string, it is the only possible header for this property
+            if (!(possibleKeys instanceof Array)) oneLevelKey = possibleKeys as string;
```

Comment:
Is array non ?  Pas sur que ça marche le instanceOf ?

Reply:
J'ai repris le code précédent sur le test mais je peux changer